lib/questions.rb

```diff
@@ -1,226 +1,292 @@
+  # which is greater than five. e.g.
+  # [1, 3, 5, 4, 1, 2, 6, 2, 1, 3, 7]
+  # becomes [1, 3, 5, 4, 1, 2]
+  def get_elements_until_greater_than_five(array)
+  end
+
+  # turn an array (with an even number of elements) into a hash, by
+  # pairing up elements. e.g. ['a', 'b', 'c', 'd'] becomes
+  # {'a' => 'b', 'c' => 'd'}
+  def convert_array_to_a_hash(array)
+    hash = Hash[*array]
+  end
+
+  # get all the letters used in an array of words and return
+  # it as a array of letters, in alphabetical order
+  # . e.g. the array ['cat', 'dog', 'fish'] becomes
+  # ['a', 'c', 'd', 'f', 'g', 'h', 'i', 'o', 's', 't']
+  def get_all_letters_in_array_of_words(array)
+  end
+
+  # swap the keys and values in a hash. e.g.
+  # {'a' => 'b', 'c' => 'd'} becomes
+  # {'b' => 'a', 'd' => 'c'}
+  def swap_keys_and_values_in_a_hash(hash)
+    hash.invert
+  end
+
+  # in a hash where the keys and values are all numbers
+  # add all the keys and all the values together, e.g.
+  # {1 => 1, 2 => 2} becomes 6
+  def add_together_keys_and_values(hash)
+    array = hash.to_a.flatten!
+    array.inject {|sum, elem| sum + elem}
+  end
+
+  # take out all the capital letters from a string
+  # so 'Hello JohnDoe' becomes 'ello ohnoe'
+  def remove_capital_letters_from_string(string)
+    string.delete "HJD"
+  end
+
+  # round up a float up and convert it to an Integer,
+  # so 3.214 becomes 4
+  def round_up_number(float)
+    float.ceil
+  end
+
+  # round down a float up and convert it to an Integer,
+  # so 9.52 becomes 9
+  def round_down_number(float)
+    float.floor
+  end
+
+  # take a date and format it like dd/mm/yyyy, so Halloween 2013
+  # becomes 31/10/2013
+  def format_date_nicely(date)
+  end
+
+  # get the domain name *without* the .com part, from an email address
+  # so alex@makersacademy.com becomes makersacademy
+  def get_domain_name_from_email_address(email)
+    domain = email.gsub(/.+@([^.]+).+/, '\1')
+  end
+
+  # capitalize the first letter in each word of a string,
+  #  except 'a', 'and' and 'the'
+  # *unless* they come at the start of the start of the string, e.g.
+  # 'the lion the witch and the wardrobe' becomes
+  # 'The Lion the Witch and the Wardrobe'
+  def titleize_a_string(string)
+    title_cases = Hash.new {|_,k| k.capitalize }.merge({'And' => 'and', 'The' => 'the', 'A' => 'a'})
+    new_string  = string.split(" ").map {|w| title_cases[w] }.join(' ')
+    new_string.capitalize
+  end
+
+  # return true if a string contains any special characters
+  # where 'special character' means anything apart from the letters
+  # a-z (uppercase and lower) or numbers
+  def check_a_string_for_special_characters(string)
+    /\W/ === string
+  end
+
+  # get the upper limit of a range. e.g. for the range 1..20, you
+  # should return 20
+  def get_upper_limit_of(range)
+    range.last
+  end
+
+  # should return true for a 3 dot range like 1...20, false for a
+  # normal 2 dot range
+  def is_a_3_dot_range?(range)
```

Comment:
Unused method argument - `range`. If it's necessary, use `_` or `_range` as an argument name to indicate that it won't be used. You can also write as `is_a_3_dot_range?(*)` if you want the method to accept any arguments but don't care about them.<br>Rename `is_a_3_dot_range?` to `a_3_dot_range?`.